src/django/api/reports.py

```diff
@@ -22,11 +29,80 @@ def monthly_promoted_name_and_address():
                     month = h[1].strftime('%Y-%m')
                     data[month] = data.get(month, 0) + 1
     return [['month', 'facilities promoted'],
-            sorted(data.items(), key=lambda x: x[0])]
+            sort_by_first_column(data.items())]
+
+
+def geocoding_time_without_queue(date_format):
+    temp_data = dict()
+    listitems = FacilityListItem.objects.values_list(
+        'created_at', 'processing_results')
+    for f in listitems:
+        for r in f[1]:
+            if r['action'] == ProcessingAction.GEOCODE:
+                started = datetime.strptime(r['started_at'],
+                                            '%Y-%m-%d %H:%M:%S.%f')
```

Comment:
We use the `%Y-%m-%d %H:%M:%S.%f` a few times across reports so consider extracting this as a constant.

Reply:
I grabbed the other date formats from this file as well, since it seems likely we'll keep using each of them.